### PR TITLE
Rustup to rustc 1.36.0-nightly (13fde05b1 2019-05-03)

### PR DESCRIPTION
**Pull request by Manishearth**
Trying to deal with changes from https://github.com/rust-lang/rust/pull/60462

Moved from https://github.com/rust-lang/rust-clippy/pull/4060 so everyone can collab on the rustup branch.

### CODE REVIEW

**Comment by Manishearth** [comment]
oops, forgot to make it a draft. oh well.

**Comment by Manishearth** [comment]
It builds now

**Comment by Manishearth** [comment]
Passes tests too. r? @oli-obk @eddyb

File: clippy_lints/src/utils/mod.rs [hunk 1]
```diff
@@ -213,7 +213,7 @@ pub fn match_path_ast(path: &ast::Path, segments: &[&str]) -> bool {
 }
 
 /// Gets the definition associated to a path.
-pub fn path_to_def(cx: &LateContext<'_, '_>, path: &[&str]) -> Option<def::Def> {
+pub fn path_to_res(cx: &LateContext<'_, '_>, path: &[&str]) -> Option<(def::Res)> {
```
Comment (eddyb) [inline clippy_lints/src/utils/mod.rs]: Oops I had forgotten to comment about the strange parens in the return type.

**Pull request merged**